Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Sebastien Loriot, Maxime Gimeno
+
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+#define CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+
+#include <CGAL/Polygon_mesh_processing/orientation.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/boost/graph/named_function_params.h>
+#include <CGAL/boost/graph/copy_face_graph.h>
+#include <CGAL/Kernel_traits.h>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <vector>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace extrude_impl{
+
+template<typename PMAP, typename Vector>
+struct Const_dist_translation{
+  Const_dist_translation(PMAP map, const Vector& dir, const double d)
+    :map(map), dir(dir), d(d){}
+  
+  template<typename VertexDescriptor, typename U>
+  void operator()(const VertexDescriptor vd, const U&)
+  {
+    typename boost::property_traits<PMAP>::value_type p = get(map, vd) + d*dir;
+    put(map, vd, p);
+  }
+  
+  PMAP map;
+  Vector dir;
+  double d;
+};
+
+struct Identity_functor
+{
+  template<typename T, typename U>
+  void operator()(const T&, const U&){}
+};
+}//end extrude_impl
+
+/**
+ * \ingroup PMP_meshing_grp
+ * Extrudes the open mesh input and put the result in output. The mesh generated is a closed mesh with 
```

Comment:
lower case

Reply:
"put" should be "puts"

Reply:
missing back-quotes on `output`

Reply:
"open mesh" and "closed mesh" should mention "surface mesh"